src/modules/problemMethods.ts

```diff
@@ -59,6 +61,16 @@ const constructFindAllProblemsWhereFilter = async (
         where.id.notIn = data.excludeProblemIds;
     }
 
+    if (!admin && !problemEditor) {
+        if (where.AND && Array.isArray(where.AND)) {
+            where.AND.push({ OR: [{ authorId: userId }, { archived: false }] });
+        } else if (where.AND) {
+            where.AND.OR = [{ authorId: userId }, { archived: false }];
+        } else {
+            where.AND = { OR: [{ authorId: userId }, { archived: false }] };
+        }
+    }
+
```

Comment:
I think we can reduce code difficulty if declare AND as Array at the beginnig of function

``` ts
const where = {
        AND: []
    } as Prisma.ProblemWhereInput;
```